src/core/common/timer.hpp

```diff
@@ -215,6 +215,44 @@ class TimerMilli : public Timer
     TimerMilliScheduler &GetTimerMilliScheduler(void) const;
 };
 
+/**
+ * This class implements a millisecond timer which also tracks an arbitrary context information.
```

Comment:
suggestion: "which also tracks an arbitrary context information" -> "that also maintains a user context pointer"